### PR TITLE
Add simple API for a more

### PR DESCRIPTION
**Pull request by pyth0n1c**
Add a simple was to get a contentctl config
and run contentctl operations without going through
the Command Line Interface/main function in contentctl.

Assuming that you are already in a repo that has contentctl.yml, the following should work:
```
from contentctl import api
from contentctl.contentctl import validate_func

#loads by default from ./contentctl.yml, but that can be overridden. 
#config is a dict that overrides default values in that file

cfg = api.configFromFile(config={'enrichments':True})
director_obj = validate_func(cfg)

#director_obj has all your validated, constructed python objects (if you try to interactively print this out, you will get an error. that is expected)
```

### CODE REVIEW

**Comment by josehelps** [comment]
Nice thank you for the quick turn around. Waiting on CI but LGTM.

**Review by josehelps — APPROVED** [review]

File: contentctl/api.py
Comment (cmcginley-splunk) [inline contentctl/api.py:3]: Can this be removed?

File: contentctl/api.py
Comment (cmcginley-splunk) [inline contentctl/api.py:5]: `test_common` is unused

File: contentctl/api.py
Comment (cmcginley-splunk) [inline contentctl/api.py:9]: nit: not something to be addressed in this MR since I know it extends beyond this file, but we should consider going with more traditional PEP8 Python style guidelines and use snake-case for function definitions

File: contentctl/api.py
Comment (cmcginley-splunk) [reply contentctl/api.py:5]: nit: again, not something for this MR, but just calling it out here, but, on the flip-side, class names should not be in snake-case for PEP8

File: contentctl/api.py
Comment (cmcginley-splunk) [inline contentctl/api.py:112]: nit: there should be a space after `#`

File: contentctl/api.py
Comment (cmcginley-splunk) [inline contentctl/api.py:55]: Could the return type not just be `test_common` since `test` and `test_servers` both inherit `test_common`?

**Comment by cmcginley-splunk** [comment]
Could I similarly call `test_common_func(cfg)`?

File: contentctl/api.py
Comment (pyth0n1c) [reply contentctl/api.py:9]: We discussed improving our style guidelines/requirements last week.
During reviews, let's start "requiring" that new functions/classes follow guidelines.
Fixing existing ones to follow those guidelines is a bit more involved, but there's no good reason not to at least start enforcing it for new things :)

Comment (cmcginley-splunk) [reply contentctl/api.py:9]: Agreed! Yeah I think if we're onboard to do PEP8 as a style guide, then doing it as we go forward is a good idea :)

File: contentctl/api.py
Comment (pyth0n1c) [reply contentctl/api.py:3]: Yes, this has been removed 👍

File: contentctl/api.py
Comment (pyth0n1c) [reply contentctl/api.py:5]: I removed test_common. You are correct about following Pep-8. I have addressed this for function names in the new file, but have not addressed it for imported classes.
For now, we will get better at new code following proper conventions.  We will eventually have an MR that cleans up existing code as well.

File: contentctl/api.py
Comment (pyth0n1c) [reply contentctl/api.py:112]: Agreed, there were a few other comments lacking this space as well in the file.

File: contentctl/api.py
Comment (pyth0n1c) [reply contentctl/api.py:55]: Yes - both functions that return configs now return a test_common.
We will leave it up to type-checking in the IDE + users to ensure that if they are accessing fields unique to a test or test_servers object that they have requested (and returned) the correct object type.

**Comment by pyth0n1c** [comment]
> Could I similarly call `test_common_func(cfg)`?

Yes, this should work too. I moved the creation of the container objects (part of the `test` object, not the underlying actual containers themselves) from `main` into `test_common_func` to account for this:
https://github.com/splunk/contentctl/blob/bf63cdd76dc8992fc41dfafa20ca8dd8ab3752ad/contentctl/contentctl.py#L105-L108

Doing a bit more logic in the class's pydantic validator functions might be better for this, but we can chat about that.

**Pull request merged**
